test/integration/AirBroSMCampaign/AirBroExistingTokenNftDropSMCampaign.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { MerkleTree } from "merkletreejs"
+const { keccak256 } = ethers.utils
+
+
+export function AirbroFactorySMCampaignShouldAirDropExistingToken(){
+    it('should fund and drop tokens to newly minted nft1155 holders',async function(){
+        const totalAirdropAmount = ethers.utils.parseEther("1000");
+        const tokensPerClaim = 100;
+        const airdropDurationInDays = 30 //days
+        
+        // minging the admin 1000 of the existing tokens
+        await this.testToken.connect(this.signers.deployer).mint(this.signers.deployer.address, totalAirdropAmount);
+        
+        // creating new airdrop
+        const newAirdrop = await this.airbroFactorySMCampaign.connect(this.signers.deployer).dropExistingTokensToNftHolders(
+            this.test1155NftCollection.address, // rewardedNftCollection,
+            tokensPerClaim,
+            this.testToken.address, //existing token address
+            totalAirdropAmount, // total tokens to be rewarded
+            airdropDurationInDays,
+            )
+            
+            expect(newAirdrop).to.emit(this.airbroFactorySMCampaign, "NewAirdrop");
+            
+            
+            const existingDropFactory = await ethers.getContractFactory("ExistingTokenDropSMCampaign");
+            const tokenDropContract = existingDropFactory.attach(await this.airbroFactorySMCampaign.airdrops(0)); // Address of newly created airdrop. How will this be sent to the frontend ?
```

Comment:
Solution 1. Listen to NewAirdrop event and get information from there
Solution 2. Create a subgraph https://thegraph.com/docs/en/developer/quick-start/ and use it